apps/hash-graph/lib/graph/src/api/rest/json_schemas/entity_type.json

```diff
@@ -10,63 +10,150 @@
       ]
     },
     "kind": {
+      "type": "string",
       "enum": ["entityType"]
     },
     "$id": {
-      "type": "string",
-      "format": "uri"
+      "$ref": "#/definitions/VersionedUrl"
     },
     "type": {
+      "type": "string",
       "enum": ["object"]
     },
-    "title": { "type": "string" },
-    "description": { "type": "string" },
+    "title": {
+      "type": "string"
+    },
+    "description": {
+      "type": "string"
+    },
+    "allOf": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/EntityTypeReference"
+      }
+    },
     "examples": {
-      "$comment": "Example Entity instances",
       "type": "array",
       "items": {
-        "type": "object",
-        "propertyNames": {
-          "$comment": "Property names must be a valid URL to a Property Type",
-          "type": "string",
-          "format": "uri"
-        }
+        "type": "object"
       }
     },
-    "properties": { "$ref": "#/$defs/propertyTypeObject" },
+    "properties": {
+      "$ref": "#/definitions/PropertyTypeObject"
+    },
     "required": {
       "type": "array",
       "items": {
-        "type": "string",
-        "format": "uri"
+        "$ref": "#/definitions/BaseUrl"
       }
     },
-    "links": { "$ref": "#/$defs/linkTypeObject" }
+    "links": {
+      "type": "object",
+      "propertyNames": {
+        "$ref": "#/definitions/VersionedUrl"
+      },
+      "patternProperties": {
+        "^.*$": {
+          "type": "object",
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum": ["array"]
+            },
+            "ordered": {
+              "type": "boolean",
+              "default": false
+            },
+            "items": {
+              "$ref": "#/definitions/OneOfEntityTypeReference"
+            },
+            "minItems": {
+              "type": "integer",
+              "minimum": 0
+            },
+            "maxItems": {
+              "type": "integer",
+              "minimum": 0
+            }
+          },
+          "required": ["type", "ordered", "items"],
+          "additionalProperties": false
+        }
+      }
+    }
   },
   "additionalProperties": false,
-  "required": ["$schema", "kind", "$id", "type", "title", "properties"],
-  "$defs": {
-    "propertyTypeObject": {
+  "required": ["$schema", "kind", "type", "$id", "title", "properties"],
+  "definitions": {
```

Comment:
Can we please keep using `$defs`?

Reply:
I'd like that, but `$defs` has the problem that types that are not referenced explicitly get omitted and then result in `object`. `definitions` is due to OpenAPI special cased and will always be included.
I can create a repro to show it of if needed.

Reply:
No, see https://github.com/hashintel/hash/pull/2838#issuecomment-1660700773; this error only happens if we use `$defs` not `definitions`. `PropertyValues` is not directly referenced in the "parent" schema, only in `$defs`, and therefore not included (because `definitions` is special-cased), but if it is in `definitions`, it is included regardless.

Reply:
OpenAPI 3.0 is based on `draft-04`, which doesn't yet have `$defs` and instead uses `definitions` (https://datatracker.ietf.org/doc/html/draft-fge-json-schema-validation-00#section-5.5.7)